cuda/matrix/ell_kernels.cu

```diff
@@ -39,23 +39,52 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/base/types.hpp>
 
```

Comment:
You also need to include the tuple header (for `std::tuple`):
```C++
#include <tuple>
```
Before the `#include <ginkgo/*>` statements but after `#include "core/matrix/ell_kernels.hpp"` according to the guidelines (github does not allow me to comment there).

Reply:
Should include `array` when using `std::array` instead of `std::tuple`

Reply:
done